sycl/include/sycl/ext/oneapi/matrix/static-query.hpp

```diff
@@ -96,22 +106,29 @@ struct tpu_params<tpu::amx, void, void, void, M, N, K> {
     uint32_t nsize;
     uint32_t ksize;
   };
-  static constexpr int num_combinations = 2;
+  static constexpr int num_combinations = 5;
```

Comment:
Not a request to change. 
In my opinion, the code would look nicer, more compact and you would not need that 'num_combinations' var if had something like this:
  using mt = matrix_type;
  combination combinations[] = {
      {16, 16, 64, mt::sint8, mt::sint8, mt::sint32},
      {16, 16, 64, mt::sint8, mt::uint8, mt::sint32},
      {16, 16, 64, mt::uint8, mt::sint8, mt::sint32},
      {16, 16, 64, mt::uint8, mt::uint8, mt::sint32},
      {16, 16, 32, mt::bf16, mt::bf16, mt::fp32}
  };

Reply:
@v-klochkov It is a nice way to initialize the array. Also, we won't need a constructor in this case. I mainly added the constructor to be able to return these sizes as constexpr. But I tried your suggestion and it also works.
However, num_combination is still needed as the user would want to know how many combinations the hardware accepts.   
  static constexpr int num_combinations = 5;
  using mt = matrix_type;
  static constexpr combination combinations[num_combinations] = {
      {16, 16, 64, mt::sint8, mt::sint8, mt::sint32},
      {16, 16, 64, mt::sint8, mt::uint8, mt::sint32},
      {16, 16, 64, mt::uint8, mt::sint8, mt::sint32},
      {16, 16, 64, mt::uint8, mt::uint8, mt::sint32},
      {16, 16, 32, mt::bf16, mt::bf16, mt::fp32}
  };

Reply:
Can we derive `num_combinations` from the size of the array?  Something like:

```c
static constexpr int num_combinations = sizeof(combinations) / sizeof(combination);
```

Reply:
Yes, that works as well. I will make the change.